### PR TITLE
adding the ability to whitelist specific CVEs for security tests

### PR DESCRIPTION
**Pull request by sharifelgamal**

### CODE REVIEW

File: security_check/main.py [hunk 1]
```diff
@@ -23,6 +23,11 @@
     _CRITICAL: 3,
 }
 
+try:
+  _WHITELIST = json.load(open('whitelist.json', 'r'))
```
Comment (dlorenc) [inline security_check/main.py]: Could the path to this be an argument to the script?

File: security_check/main.py [hunk 2]
```diff
@@ -39,6 +44,8 @@ def _check_image(image, severity):
     for vuln in parsed.get('vulz_analysis', []):
         if vuln.get('patch_not_available'):
             continue
+        if vuln.get('vulnerability') in _WHITELIST:
```
Comment (dlorenc) [inline security_check/main.py]: Probably worth logging the whitelist before doing anything.

File: security_check/main.py (hunk 1 above)
Comment (sharifelgamal) [inline security_check/main.py]: done

File: security_check/main.py (hunk 2 above)
Comment (sharifelgamal) [inline security_check/main.py]: Done.

File: security_check/main.py [hunk 3]
```diff
@@ -30,15 +30,23 @@ def _run_gcloud(cmd):
     return json.loads(output)
 
 
-def _check_image(image, severity):
+def _check_image(image, severity, whitelist_file):
```
Comment (dlorenc) [inline security_check/main.py]: Small nit: could you parse the file in main and pass in the whitelist?

Comment (sharifelgamal) [inline security_check/main.py]: Done.

File: security_check/main.py [hunk 4]
```diff
@@ -39,6 +39,8 @@ def _check_image(image, severity):
     for vuln in parsed.get('vulz_analysis', []):
         if vuln.get('patch_not_available'):
             continue
+        if vuln.get('vulnerability') in whitelist:
+            continue
```
Comment (dlorenc) [inline security_check/main.py]: Can we log this?

**Pull request merged**